### ISSUE TITLE
Admin not showing https certificates/keys

### ISSUE THREAD

**Issue by poldiderjungdrache** [ioBroker/ioBroker.admin#1214]
**Describe the bug**  
I am in the process of setting up HTTPS for iobroker.
I have added certificate and the corresponding private key to iobroker by linking to the path to the files in FS.
<img width="1641" alt="iobroker certs" src="https://user-images.githubusercontent.com/59320185/143500518-a8769517-9a56-40ac-87f5-bf6b69e08b0b.png">

I saved the change and then I try to add them in admin. However they don't show up in the dropdown list:
<img width="1513" alt="admin cert" src="https://user-images.githubusercontent.com/59320185/143500693-97570ae1-7d65-45ad-9b4c-08ce7a581ca8.png">

However in the web adapter configuration they are correctly shown:
<img width="1288" alt="web" src="https://user-images.githubusercontent.com/59320185/143500788-901af85e-e62a-4965-8c7c-1140bcfe3864.png">

I can select them in the dropdown and restart web. Web comes up and in the browser the correct cert is used. So I don't think this has anything to do with access to the files in FS or am I missing something?

**To Reproduce**  
1. Go to System Settings
2. Go to Certificates
3. Add entry
4. Copy path to cert or private key
5. Try to use either of the above in admin config.

**Expected behavior**  

**Versions:**  
 - Adapter version: 5.1.28
 - JS-Controller version: 3.3.20
 - Node version: 12.22.7
 - Operating system: Raspbian Buster

**Comment by Apollon77** [comment]
Can you restar tadmin adapter please ... do they show up then?

**Comment by poldiderjungdrache** [comment]
Actually I did some more testing last night.
I added cert and key in 3 different ways (1. pull in from FS, 2. drag and drop the pem, 3. copy the cert and key from inside the pem):
<img width="1803" alt="2021-11-26_09-17-18" src="https://user-images.githubusercontent.com/59320185/143549315-a4e0f1ff-bb64-4017-b4f7-478aa06c73b5.png">

Only 2 and 3 show up in admin (even after full restart of iobroker and host):
<img width="1864" alt="2021-11-26_09-16-42" src="https://user-images.githubusercontent.com/59320185/143549522-e9baf1f0-1787-42d4-b927-7df5c1f1d96b.png">

**Comment by JB1985** [comment]
And the imported Certs are working?

**Comment by poldiderjungdrache** [comment]
I am using the drag and drop ones (2) at the moment and admin is fine with that.
Copy and paste (3) I haven’t tried yet but I would expect them to be fine as well. Can try later.
Simon

**Comment by Apollon77** [comment]
See question in other issue two (whxy we jhave two now?) ... "copy&paste" ... how you copy? Seems the line breaks gets missing when you do that

**Comment by JB1985** [comment]
> (whxy we jhave two now?)

Why it is not the same. Here is the problem that the certs on the filesystem /opt/certs/*.pem are not shown or available in admin.0.

In #1212 the issue is the manually import of the certifcates.

**Comment by poldiderjungdrache** [comment]
Sorry but I think we are mixing up issues here.
I have no issues with the cert or key per se. They work fine. Admin starts and no issues in the log files.
However I have issues with cert and key added via path. The issue is that they don't show up in admin, i.e. there is no entry in the pull down menu.
Please check the screen shots carefully.
In admin certificate pull down menu I would expect 3 entries, i.e. certificate_1, cert2 and certificate_3. However certificate_1 (and key_1) is missing in the pull down. 
in web adapter config certificate_1 is available (see screen shot at the beginning). 
Hope that helps to clarify things.

**Comment by rrov1** [comment]
I have the same problem:
Versions:
Adapter version: admin:5.2.3, web:3.4.9
JS-Controller version: 3.3.22
Node version: 14.18.2
Operating system: debian 11

**Comment by poldiderjungdrache** [comment]
Yes issue is still there in 5.2.3. But I also did not see anything in the changelog that would suggest otherwise.

**Comment by poldiderjungdrache** [comment]
Interestingly [SirHis](https://github.com/ioBroker/ioBroker.admin/issues/1212#issuecomment-991293950) appeared to have got it working just the way I would like to.
Maybe I will try to re-do my whole cert/key config one more time from scratch later today.

**Comment by rrov1** [comment]
@poldiderjungdrache Thanks for your hint, but I don't get it to work. I've added private/public to the path, after that I was able to choose the certificates in the admin configuration, but admin does not start. I get an error: 
`Cannot create web-server: Error: error:0909006C:PEM routines:get_name:no start line`

**Comment by rrov1** [comment]
@poldiderjungdrache Yes, I can confirm that the error sill occurs with admin v5.3.1.

**Comment by poldiderjungdrache** [comment]
hi,
honestly I hadn't update to 5.3.1 but I did so today.
I must say that for me the cert and key are showing up now (to my surprise):
![Screenshot 2022-02-28 at 22 25 48](https://user-images.githubusercontent.com/59320185/156061263-5bb978e0-ecbd-4827-8451-fbe2623d7828.png)

and below the screenshot from my base settings:
![Screenshot 2022-02-28 at 22 27 13](https://user-images.githubusercontent.com/59320185/156061475-ce27c50f-3c1b-46a4-a582-969c9ced1352.png)

With this setting the admin starts without issues and ssl is working as expected.

**Comment by rrov1** [comment]
@poldiderjungdrache I have no luck. I removed the certificates (+save), changed language to english, added an additional instance of admin. Then I readded the certificates in configuration (under a different name) but in admin.1 there is no way to choose them :-(.

**Comment by rrov1** [comment]
Hi, 
me again. Today I found the solution in the iobroker forum: 
https://forum.iobroker.net/topic/51592/zertifikate-nicht-mehr-%C3%BCber-dateipfad-ausw%C3%A4hlbar
the names for the certificate need to end with (or contain - not checked) with "Public" or "Private". After renaming, I was able to to choose the certificates in admin.
regards,
rrov1

**Comment by poldiderjungdrache** [comment]
> Hi,
> me again. Today I found the solution in the iobroker forum:
> https://forum.iobroker.net/topic/51592/zertifikate-nicht-mehr-%C3%BCber-dateipfad-ausw%C3%A4hlbar
> the names for the certificate need to end with (or contain - not checked) with "Public" or "Private". After renaming, I was able to to choose the certificates in admin.
> regards,
> rrov1

True I had forgot to mention that I added the Public/Private to my cert names a while back because I read the same post. That did not seem to help immediately though. Maybe the admin adapter update helped but I am not sure.

**Comment by petterpet** [comment]
I set up SSL today and found out the following:

- I had to copy the certificate file to the host and enter the absolut path in the system settings. Copy&Paste did not work - then the adapters not come up again.
- To see new certificates in the dropdown reloading of the admin-website is required (but no restart of the adapter).
- To see new certificates in the admin-adapter the names must end with "Public" / "Private". Or use the old UI by enabling the checkbox in the admin-adapter-settings.
- For other adapters like web or nodered the name is not relevant.
